tests/test_file_watch.rb

```diff
@@ -55,6 +55,25 @@ def test_events
       assert($deleted)
       assert($unbind)
     end
+
+    # Refer: https://github.com/eventmachine/eventmachine/issues/512
+    def test_invalid_signature
+      # This works fine with kqueue, only fails with linux inotify.
+      omit_if(EM.kqueue?)
+
+      EM.run {
+        file = Tempfile.new('foo')
+
+        w1 = EventMachine.watch_file(file.path)
+        w2 = EventMachine.watch_file(file.path)
```

Comment:
The watcher it returns is valid but the binding to watch map that is used to keep reference of the watch is not properly updated, so stopping the watch raises this error.

The inotify and kqueue behaviour is reported more thoroughly in #512.